pkg/instancegroups/scaleup.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instancegroups
+
+import (
+	"context"
+
+	api "k8s.io/kops/pkg/apis/kops"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// scaleProvider session data
+type scaleProvider struct {
+	GroupUpdate *RollingUpdateInstanceGroup
+}
+
+// NewScaleUpRollout creates and returns a scale provider
+func NewScaleUpRollout(update *RollingUpdateInstanceGroup) Rollout {
+	return &scaleProvider{GroupUpdate: update}
+}
+
+// RollingUpdate performs a duplicate rolling update of an instance group
+func (p *scaleProvider) RollingUpdate(ctx context.Context, list *api.InstanceGroupList) error {
+	group := p.GroupUpdate.CloudGroup.InstanceGroup
+	name := group.Name
+	strategy := group.Spec.Strategy
+	update := p.GroupUpdate.Update
+
+	// @step: increase the size of the instancegroup by x percent
+	var newMinSize, newMaxSize, oldMinSize, oldMaxSize int32
+	total := int32(p.GroupUpdate.CloudGroup.Size())
+	oldMinSize = int32(p.GroupUpdate.CloudGroup.MinSize)
+	oldMaxSize = int32(p.GroupUpdate.CloudGroup.MaxSize)
+	newMinSize = int32(p.GroupUpdate.CloudGroup.Size() * 2)
+	newMaxSize = int32(p.GroupUpdate.CloudGroup.MaxSize)
+	if newMinSize > newMaxSize {
+		newMaxSize = newMinSize
```

Comment:
I'm missing something here.  If min = max , won't this leave the cluster unchanged in size?

Reply:
I'm basically using the min as a desired count here .. so i double the original min and if the new max size doesn't support it we set them equal ..